src/AutoRest.CSharp/LowLevel/Decorator/MethodAmbiguityResolver.cs

```diff
@@ -0,0 +1,169 @@
+                    if (IsSameMethod(clientMethod, clientMethodSourceModel))
+                    {
+                        return clientMethod;
+                    }
+                    ambiguityFound = true;
+                }
+            };
+
+            if (ambiguityFound)
+            {
+                return clientMethod.CloneWithAllParametersRequired();
+            }
+            return clientMethod;
+        }
+
+        /// <summary>
+        /// Check if a client method is ambiguous with the method in source model.
+        /// Two methods are considered as ambiguous when:
+        /// - both has the same number of required parameters
+        /// - for each required parameter, either they should be the same type in source model.
+        /// </summary>
+        /// <param name="clientMethod"></param>
+        /// <param name="methodSymbol"></param>
+        /// <returns></returns>
+        private static bool IsAmbiguousMethod(LowLevelClientMethod clientMethod, IMethodSymbol methodSymbol)
+        {
+            var clientMethodRequiredParameters = clientMethod.Signature.Parameters.Where(p => !p.IsOptionalInSignature).ToList();
+            var requiredParameterSymbols = methodSymbol.Parameters.Where(p => !p.HasExplicitDefaultValue).ToList();
+            if (clientMethodRequiredParameters.Count != requiredParameterSymbols.Count)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < clientMethodRequiredParameters.Count; i++)
+            {
+                if (!IsSameParameter(clientMethodRequiredParameters[i], requiredParameterSymbols[i]))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Check if a clientMethod has the same paramters as a method from source model has.
+        /// Both methods should:
+        /// - have the same number of parameters
+        /// - each parameter has the same type
+        /// - each parameter has the same default value (e.g. optional or required)
+        /// </summary>
+        /// <param name="clientMethod"></param>
+        /// <param name="clientMethodSourceModel"></param>
+        /// <returns></returns>
+        private static bool IsSameMethod(LowLevelClientMethod clientMethod, IMethodSymbol clientMethodSourceModel)
+        {
+            if (clientMethod.Signature.Parameters.Count != clientMethodSourceModel.Parameters.Length)
+            {
+                return false;
+            }
+            for (int i = 0; i < clientMethod.Signature.Parameters.Count; i++)
+            {
+                if (!IsSameParameter(clientMethod.Signature.Parameters[i], clientMethodSourceModel.Parameters[i]))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static bool IsSameParameter(Parameter parameter, IParameterSymbol parameterSourceModel)
```

Comment:
IsParameterSignatureSame?

Reply:
I'd prefer `IsParameterTypeSame`, since we don't check the name of the parameter.

Reply:
> > > The test case location is a little bit confusing. In my opinion, it should be in the HLC folder.
> > 
> > 
> > I tend not to be restricted in HLC, since theoretically the `SourceInputModel` will include any code file except for those under `Generated` folder. If you check the `MetricsAdvisor`, the HLC clients are located at the project root folder. Besides, as I mentioned before, developers could have written customization codes. So here I just put the existing codes where the Roslyn can find and parse.
> 
> I mean, for example, the namespace should not end with `_LowLevel` and the project name should not end with `_LowLevel`. In my understanding, our functionality is: starting from HLC, we generate such LLC. So it seems the test project should start from "without LLC".

- If the original codes are not in the `_LowLevel` namespace, then there won't be ambiguity. I could have a better name for this, but I think that won't affect the readiness.
- The `_LowLevel` is to identify the test scenario, not to describe a real customer library. I think it would be easier to locate the test project.